### ISSUE TITLE
[Bug]: OneSignal.User.addTagWithKey method crash app in IOS when pass boolean

### ISSUE THREAD

**Issue by rafaelbarbosadrip** [OneSignal/OneSignal-Flutter-SDK#728]
### What happened?

When we call the method `OneSignal.User.addTagWithKey` passing the boolean value the app happen crash (close suddenly)

### Steps to reproduce?

```Markdown
When we call the method `OneSignal.User.addTagWithKey` passing a boolean like this example:


OneSignal.User.addTagWithKey('anyKey',true);
```

### What did you expect to happen?

Set the tag and not crash the app.

### OneSignal Flutter SDK version

5.0.0

### Which platform(s) are affected?

- [X] iOS
- [ ] Android

### Relevant log output

```Shell
Could not cast value of type '__NSCFBoolean' (0x10c8ba3b8) to 'NSString' (0x10ccb2b48)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by emawby** [comment]
Thank you for reporting this is fixed in #733

**Comment by emawby** [comment]
Closing with the release of 5.0.1
